src/main/java/seedu/address/model/person/Allergy.java

```diff
@@ -0,0 +1,66 @@
+package seedu.address.model.person;
+
+import seedu.address.model.tag.Tag;
+
+import static java.util.Objects.requireNonNull;
+import static seedu.address.commons.util.AppUtil.checkArgument;
+
+/**
+ * Represents an allergy in the address book.
+ * Guarantees: immutable; name is valid as declared in {@link #isValidAllergy(String)}
+ */
+public class Allergy {
+
+    public static final String MESSAGE_CONSTRAINTS = "Allergy names should be alphanumeric";
+
+    public static final String VALIDATION_REGEX = "\\p{Alnum}+";
```

Comment:
I was wondering if this should be the validation_regex for allergy
"/*
     * The first character of the address must not be a whitespace,
     * otherwise " " (a blank string) becomes a valid input.
     */
    public static final String VALIDATION_REGEX = "[\\p{Alnum}][\\p{Alnum} ]*";
    "
    As seen from the implementation in Name Class